gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/MatchStep.java

```diff
@@ -362,7 +360,7 @@ private boolean hasPathLabel(final Path path, final Set<String> labels) {
         while (true) {
             if (this.first) {
                 this.first = false;
-                this.initializeMatchAlgorithm(TraversalEngine.Type.STANDARD);
+                this.initializeMatchAlgorithm(true);
```

Comment:
Shouldn't this be `false`?

Reply:
oh wow - good catch. i guess the tests pass either way though...don't know if that's bad.???

Reply:
It's not bad. I don't fully understand the code in `CountMatchAlgorithm` (the only place where the boolean flag makes a difference), but it seems like it only reorders the match traversals, possibly to optimize memory consumption and/or processing time.

Reply:
that makes sense - fixed and pushed